rep-ixxxx.rst

```diff
@@ -0,0 +1,826 @@
+There is no support for joint velocity, acceleration or effort, nor a group identifier or index.
+The message size is fixed and the maximum number of joints supported is ten (``10``).
+
+Early server implementations also accepted this message for enqueuing trajectory points.
+This usage has been deprecated (and support removed from ``industrial_robot_client``) and it is an error for clients to try to use ``JOINT_POSITION`` for this purpose.
+Driver authors must use `JOINT_TRAJ_PT`_ and `JOINT_TRAJ_PT_FULL`_ messages instead.
+
+Note that use of this message for encoding joint states is also deprecated (because of the lack of support for motion groups, joint velocity or accelerations mentioned earlier in this section), and new server implementations are recommended to use `JOINT_FEEDBACK`_ to encode joint states.
+
+For an example bytestream, see `Example: JOINT_POSITION`_.
+
+Message type: *asynchronous publication*
+
+Assigned message identifier: 10 (``0x0A``)
+
+Status: *deprecated*
+
+Supported by generic nodes: *yes* (joint state reporting), *no* (enqueuing)
+
+Dataflow direction (typically): client → server, server → client
+
+Message::
+
+  Prefix
+  Header
+  sequence         : shared_int
+  joint_data       : shared_real[10]
+
+Notes
+
+#. The ``sequence`` field uses zero-based numbering.
+#. The ``sequence`` field is not used when reporting joint state and shall be set to zero (``0``) by server implementations.
+#. Elements of ``joint_data`` that are not used must be initialised to zero (``0``) by the sender.
+#. The size of the ``joint_data`` array is ``10``, even if the server implementation does not need that many elements (for instance because it only has six joints).
+#. Controllers that support or are configured with more than a single motion group should use the `JOINT_FEEDBACK`_ message if they wish to report joint state for all configured motion groups (note: unfortunately there is currently no support for `JOINT_FEEDBACK`_ messages in the generic clients, that will have to be added).
+#. The elements of the ``joint_data`` field shall represent the joint space positions of the corresponding joint axes of the controller.
+   In accordance with [#REP103]_, units are *radians* for revolute or rotational axes, and *meters* for prismatic or translational axes.
+
+
+JOINT_TRAJ_PT
+-------------
+
+This message may be used by clients to enqueue trajectory points for execution by the server.
+
+See `Example: JOINT_TRAJ_PT`_ for bytestream example.
+
+Message type: *synchronous service*
+
+Assigned message identifier: 11 (``0x0B``)
+
+Status: *active, in use*
+
+Supported by generic nodes: *yes*
+
+Dataflow direction (typically): client → server
+
+Request::
+
+  Prefix
+  Header
+  sequence         : shared_int
+  joint_data       : shared_real[10]
+  velocity         : shared_real
+  duration         : shared_real
+
+Reply::
+
+  Prefix
+  Header
+  dummy_data       : shared_real[10]
```

Comment:
This really exists?  Why does it have dummy data?

Reply:
Yes. At least the `fanuc_driver` Karel side does this and `abb_driver` Rapid as well.

It's not strictly needed though, so we could spec a header only reply here.

Reply:
Just curious where it originated.  Not sure we can/should change something that has been around so long.

Reply:
Addressed in eb52fcb95902a5f315a1989c86631a4828077ebc.